### ISSUE TITLE
How to filter by grade?

### ISSUE THREAD

**Issue by MasterCarl** [schul-cloud/schulcloud-content-api#7]
I have trouble figuring out how to filter content by targetGroup.grade. Querying `/contents/?query=asdf&filter[targetGroups][grade]=9`, I get a server error

**Comment by Dome-GER** [comment]
Elasticsearch stores nested objects in a special way. Therefore, your request is not correct. Nevertheless, there seems to be an issue with our nested objects. I changed the Elasticsearch index and now the corresponding Elasticsearch query is
```
POST http://localhost:9200/schulcloudcontent/_search
{
    "query": {
        "nested": {
            "path": "targetGroups",
               "query": {
                  "bool": {
                     "must": [
                         {"term": {"targetGroups.grade": "9"}}
                     ]
                 }
             }
        }
    }
}
```
or, if you are looking for all contents that are suited for the 9th grade in BY (**AND**):
```
POST http://localhost:9200/schulcloudcontent/_search
{
    "query": {
        "nested": {
            "path": "targetGroups",
               "query": {
                  "bool": {
                     "must": [
                         {"term": {"targetGroups.grade": "9"}},
                         {"term": {"targetGroups.state": "by"}}
                     ]
                 }
             }
        }
    }
}
```
The "terms" filter can be used for the **OR** condition:
```
POST http://localhost:9200/schulcloudcontent/_search
{
    "query": {
        "nested": {
            "path": "targetGroups",
               "query": {
                  "bool": {
                     "must": [
                         {"terms": {"targetGroups.grade": ["9", "8"]}}
                     ]
                 }
             }
        }
    }
}
```

We have to check if the querys are built correctly by the jsonapi-server.

**Comment by MasterCarl** [comment]
Thanks, but as @Langleu pointed out, we cannot query elasticsearch directly since the server is not reachable. Could you indeed please check the query builder in jsonapi-server?

**Comment by soldag** [comment]
Fixed in 0849fb64d. Querying /contents/?query=asdf&filter[targetGroups][grade]=9 is working now.

**Comment by MasterCarl** [comment]
Great, thanks!
